pep-0654.rst

```diff
@@ -335,37 +335,111 @@ typically either query to check if it has leaf exceptions for which some
 condition holds (using ``subgroup`` or ``split``) or format the exception
 (using the ``traceback`` module's methods).
 
-It is unlikely to be useful to inspect the individual leaf exceptions. To see
-why, suppose that an application caught an ``ExceptionGroup`` raised in an
-``asyncio.gather()`` call. At this stage, the context for each specific
+It is less likely to be useful to iterate over the individual leaf exceptions.
+To see why, suppose that an application caught an ``ExceptionGroup`` raised in
+an ``asyncio.gather()`` call. At this stage, the context for each specific
 exception is lost. Any recovery for this exception should have been performed
 before it was grouped with other exceptions into the ``ExceptionGroup`` [10]_.
 Furthermore, the application is likely to react in the same way to any number
 of instances of a certain exception type, so it is more likely that we will
 want to know whether ``eg.subgroup(T)`` is None or not, than we are to be
 interested in the number of ``Ts`` in ``eg``.
 
-If it does turn out to be necessary for an application to iterate over the
-individual exceptions of an ``ExceptionGroup`` ``eg``, this can be done by
-calling ``traverse(eg)``, where ``traverse`` is defined as follows:
+However, there are situations where it is necessary to inspect the
+individual leaf exceptions. For example, suppose that we have an
+``ExceptionGroup`` ``eg`` and we want to log the ``OSErrors`` that have a
+specific error code and reraise everything else. We can do this by passing
+a function with side effects to ``subgroup``, as follows:
 
 .. code-block::
 
-    def traverse(exc, tbs=None):
+    def log_and_ignore_ENOENT(err):
+        if isinstance(err, OSError) and err.errno == ENOENT:
+            log(err)
+            return False
+        else:
+            return True
+
+    try:
+        . . .
+    except ExceptionGroup as eg:
+        eg = eg.subgroup(log_and_ignore_ENOENT)
+        if eg is not None:
+            raise eg
+
+
+In the previous example, when ``log_and_ignore_ENOENT`` is invoked on a leaf
+exception, only part of this exception's traceback is accessible - the part
```

Comment:
```suggestion
exception, only part of this exception's traceback is accessible -- the part
```

Reply:
Ah sorry I needed rebase and may have lost it.